catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -429,8 +430,17 @@ private static URI getResourceOrDerivedUri(ResourceInfo resourceInfo, ResourceRe
     // check if the resource request included a qualifier in which case we are trying to retrieve
     // a derived resource and not the resource
     final Serializable qualifier = resourceReq.getPropertyValue(ResourceRequest.QUALIFIER);
-
-    if (qualifier == null) { // no qualifier means we are retrieving the resource
+    final String scheme = resourceInfo.getResourceUri().getScheme();
+
+    // no qualifier means we are retrieving the resource
+    // presence of a qualifier can mean different things. We interpret it based on protocol:
+    //   - content: this always means we are retrieving a derived resource.
+    //   - http(s): we assume the intent is to retrieve a derived resource. Is this always true?
+    //   - other: we assume we are retrieving the resource. This assumption is based on experience
+    //            with protocols that use the qualifier for purposes other than derived resources.
+    //            If protocols besides content and http(s) need to support derived resources, this
+    //            could be an issue.
+    if (qualifier == null || !Arrays.asList("http", "https", "content").contains(scheme)) {
```

Comment:
Update unit tests?

Reply:
Hm... same here. No unit tests for this derived resource handling logic. I can try to add something for this change, but the whole method needs tests.

Reply:
Added unit tests